doc: Minor updates on Asset Tracker v2 docs (#9253)
nrfconnect/sdk-nrf: applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst

* AVSystem's LwM2M Coiote Device Management - :ref:`server_setup_lwm2m`.
  No additional configuration is needed if the server is set up according to the linked documentation.

> You can also use the nRF Asset Tracker project, which is a single cloud solution supporting multiple cloud providers.

Reviewer: ```suggestion
You can also use the nRF Asset Tracker project that provides an example cloud implementation for multiple cloud providers.
```

See the :ref:`nRF_asset_tracker` section for more details.

.. _atv2_lwm2m_carrier_support:

> LwM2M carrier library support

Reviewer 2: I'd prefer if we could leave `LwM2M` out as it might confuse customers now that we also support an LwM2M-based cloud for sensor data. But maybe that's more of a renaming of the carrier library thing. In general, I don't see a reason that the protocol should be in the carrier library name.
cc @nordic-hani

Reviewer 3: Second that

Reviewer 3: The LwM2M wording was originally removed to prevent users from being confused when wanting to configure the application for lwm2m.
